packages/mui-material/src/Button/Button.js

```diff
@@ -88,83 +88,96 @@ const ButtonRoot = styled(ButtonBase, {
     ),
     '&:hover': {
       textDecoration: 'none',
-      backgroundColor: alpha(theme.palette.text.primary, theme.palette.action.hoverOpacity),
+      // backgroundColor: alpha(theme.palette.text.primary, theme.palette.action.hoverOpacity),
+      backgroundColor: theme.vars
+        ? `rgba(0,0,0,${theme.vars.opacity?.hover})`
```

Comment:
Actually this is not correct, and this reminded me why I was calling `createTheme`. Basically `createTheme` creates all colors necessary, and appends them in the palette, this is why I was creating dark and light theme and then was adding them in the `colorSchemes`. Anyway I will create a helper util to create the whole palette based on the input in color schemes :) cc @siriwatknp

Reply:
Agree, it should be something like:
```js
backgroundColor: theme.vars
   ? `rgba(${theme.vars.text.primaryChannel}, ${theme.vars.opacity?.hover})`
```
which `text.primaryChannel` could be auto-generated from `text.primary`. Does this makes sense?

Reply:
Yes, but this is just one example. My point was that `createTheme` auto generates lots of the values in the palette, for example using `augmentColor` which won't be possible if we don't invoke it. I will start with creating an alternative `createTheme` method that will take into account the color schemes.

Reply:
Got it.

Reply:
This is what I landed on https://github.com/mui/material-ui/pull/31138/commits/d8f73067a066bb3f7b110ac7cc2104f1e92502ba, the usage is much simpler now. Will keep pushing this further and add some tests. Let me know if it is close to what you had in mind initially

Reply:
Had to do one more push - https://github.com/mui/material-ui/pull/31138/commits/37bda7face0a149254060653c71e655af0b10c65. The typogrpahy depends on the palette, so I had to create it in the CssVarsProvider, as we have the resolved palette there. basically anything that depends on the palette will need to be created in the resolveTheme function, which makes sense I think.

Reply:
Overall, this makes sense to me because we need to generate the channel color before sending them to `CssVarsProvider`. As a side benefit, developers can create a theme in another file via this function (typed & autocomplete) and then use that theme in the App.

I think Joy will have to do the same because there might be some cases that the `channel` color is needed.

Reply:
Makes sense, I wonder if we need to create a channel colors for every color we have. I started with the ones I think makes sense, but let's see in the future..

One thing that would complicate things is customizing this with the variables, as people will need to create the channels and set them in the variables, for example: 

```jsx
<Button
  sx={{
    '--md-palette-primary-main': '#FF0000',
    '--md-palette-primary-mainChannel': '255 0 0', // the main channel for red
  }}
>
  Text
</Button>
```

I know we can have a function to make the conversion, but I am not sure how we can avoid it. Have you thought about it yet? We can do some "gymnastics" in the `sx` prop and generated the channel variables if some variable ends on `main`, or something similar, but I am not sure how it would complicate things. Maybe we can have a whitelist of variables that needs channel generation and do it automatically

Reply:
I think this is the only open doubt I have for now. Should we create the channels automatically for the `sx` prop too if a color CSS variable is being updated. Should we have a whitelist of variables we need the channels generated for? cc @siriwatknp

Reply:
> Should we create the channels automatically for the sx prop too if a color CSS variable is being updated

What's the use case, can you share a customization example on this?

Reply:
The specific example in https://github.com/mui/material-ui/pull/31138#discussion_r822637506 should be enough. I want to change the color of the button to be red, but for the hover to work I need to set up the channel too.